### ISSUE TITLE
Theme settings export skips some urls if it thinks they are files

### ISSUE THREAD

**Issue by davidscotson** [tmuras/moosh#302]
On export the code checks the value of the setting and if starts with a / and has dot in it, then it thinks it might be a file and checks for the existance of that file. If it finds a file that matches that location it adds it successfully to the export, however if it doesn't it just ignores that setting.

This would catch out any setting that is an internal url like "/badges/mybadges.php" and strip them out of the export. Probably just falling back to including it as a normal setting if no file found would work.

**Comment by tmuras** [comment]
Yes, you're right.
It may be a good idea to add a test case here. Do you have some real example of a theme with such a setting?

**Comment by davidscotson** [comment]
Adapatable theme's slider url options such as p1url and p2url would get tripped up by this if they linked to most internal pages:

https://moodle.org/plugins/theme_adaptable

<img width="573" alt="Screen Shot 2020-04-20 at 14 54 33" src="https://user-images.githubusercontent.com/7288187/79759780-08012280-8317-11ea-9707-fd1b2c7903d1.png">

If you use "/course/view.php?id=11" as the setting value for p1url, but leave p2url empty, then you'll only see the latter output in the settings xml file (i.e. p2url will be an empty xml tag, p1url will be missing)

**Comment by davidscotson** [comment]
{code}
                        foreach ($files as $f) {
                            if (!$f->is_directory()) {
                                $fh = $f->get_content_file_handle();
                                $meta = stream_get_meta_data($fh);
                                $uriparts = explode('/', $meta['uri']);
                                $hash = array_pop($uriparts);
                                $phar->addFile($meta['uri'], $hash);
                                $element->setAttribute('file', $hash);
                                $root->appendChild($element);
                            }
+                            $root->appendChild($element);
{code}

That's our fix for this, I'll try to create a PR when I get a moment.

Basically, it looks for a file when the setting starts with a / but if it doesn't find a file that matches, it just adds the string as a setting.
